l2geth/rollup/sync_service.go

```diff
@@ -418,9 +419,15 @@ func (s *SyncService) VerifierLoop() {
 	log.Info("Starting Verifier Loop", "poll-interval", s.pollInterval, "timestamp-refresh-threshold", s.timestampRefreshThreshold)
 	t := time.NewTicker(s.pollInterval)
 	defer t.Stop()
-	for ; true; <-t.C {
-		if err := s.verify(); err != nil {
-			log.Error("Could not verify", "error", err)
+
+	for {
+		select {
+		case <-t.C:
+			if err := s.verify(); err != nil {
+				log.Error("Could not verify", "error", err)
+			}
+		case <-s.ctx.Done():
+			return
 		}
```

Comment:
The call to `s.cancel` on line 411 will trigger the `Done` on 429

The problem that we faced was that the ticker polled again during shutdown and another round of `verify` happened while it was shutting down, resulting in the db being partially updated